base/test.jl

```diff
@@ -407,11 +358,27 @@ function finish(ts::DefaultTestSet)
     println()
     # Recursively print a summary at every level
     print_counts(ts, 0, align, pass_width, fail_width, error_width, total_width)
-    # Finally throw an error as we are the outermost test set
-    if total != total_pass
-        throw(TestSetException(total_pass,total_fail,total_error))
-    end
+end
 
+# store the results in the enclosing testset and the top level result stream
```

Comment:
I'm not sure what you mean. I think that's noisy because I put the `show` method above it but directly below the type declaration. Which had been the convention throughout the rest of the file.